react-web/src/pages/subscription/payment/Payment.tsx

```diff
@@ -0,0 +1,176 @@
+import { BigNum } from "@emurgo/cardano-serialization-lib-browser";
+import { fetchData } from "api/api";
+import Button from "components/Button/Button";
+import Modal from "components/Modal/Modal";
+import Toast from "components/Toast/Toast";
+import React, { useEffect, useState } from "react";
+import { useDispatch } from "react-redux";
+import { useLocation, useNavigate } from "react-router-dom";
+import { payFromWallet } from "store/slices/walletTransaction.slice";
+import { useAppSelector } from "store/store";
+import "./Payment.scss";
+import dayjs from "dayjs";
+import { Subscription } from "../Subscription.interface";
+
+
+function Payment() {
+  const { state } = useLocation();
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const { address, wallet } = useAppSelector((state) => state.auth);
+  const { error } = useAppSelector((state) => state.walletTransaction);
+  const [transactionId, setTransactionId] = useState("");
+  const [showError, setShowError] = useState("");
+  const [openModal, setOpenModal] = useState(false);
+  const [processing, setProcessing] = useState(false);
+  let currentTierPrice: any = 0;
+  let currentSubscriptionId: string = '';
+
+  const onCloseModal = () => { 
+    setOpenModal(false)
+    navigate(-1)
+  }
+
+  const handleError = (errorObj: any) => {
+    let errorMsg = ''
+    if (typeof errorObj === 'string') {
+        errorMsg = errorObj + ' Please try again.'
+    } else if (errorObj?.info) {
+        errorMsg = errorObj.info + ' Please try again.'
+    } else if (errorObj?.response?.message) {
+        errorMsg = errorObj?.response.message + ' Please try again.'
+    } else if (errorObj?.response?.data) {
+        errorMsg = errorObj.response.statusText + ' - ' + errorObj.response.data 
+    } else {
+      errorMsg = 'Something wrong occurred. Please try again later.'
+    }
+    setShowError(errorMsg);
+    const timeout = setTimeout(() => { 
+      clearTimeout(timeout); 
+      setShowError("") 
+    }, 5000)
+    setOpenModal(false)
+    console.log(errorMsg)
+}
+
+  useEffect(() => {
+    error && handleError(error)
+  }, [error])
+
+  const triggerPayment = async () => {
+    setShowError("");
+    setProcessing(true);
+    fetchData.get("/profile/current/balance").then((response) => {
+      const availableProfileBalance: number = response.data;
+      if (availableProfileBalance - currentTierPrice < 0) {
+        const min_fee = 1000000; // 1 ADA in lovelaces - min req for cardano wallet txn
+        const fee: BigNum = currentTierPrice > min_fee ? currentTierPrice : min_fee;
+        triggerTransactionFromWallet(fee);
+      } else {
+        fetchCurrentSubscription()
+      }
+    });
```

Comment:
why use `async` and after that use the traditional `then` way to compose `Promises` and functions?
```suggestion
    const response = await fetchData.get("/profile/current/balance")
    const availableProfileBalance: number = response.data;
    if (availableProfileBalance - currentTierPrice < 0) {
      const min_fee = 1000000; // 1 ADA in lovelaces - min req for cardano wallet txn
      const fee: BigNum = currentTierPrice > min_fee ? currentTierPrice : min_fee;
      triggerTransactionFromWallet(fee);
    } else {
      fetchCurrentSubscription()
    }
```

Reply:
done